File: DIA/Aula 4/Ranking/ranking.py
t = int(input())

for i in range(t):
    n,k = map(int,input().split())
    if (n == k):
        print(1)
    
    else:
        base = 1
        
        for i in range(n,k,-1):
            base *= i
            
            if (base > 1000000007):
                base = base % 1000000007
        
        print(base)

File: DIA/Aula 4/Ranking/ranking.c
#include <stdio.h>

int main() {
    int k,n;
    int t;
    scanf("%d",&t);

    for (int i=0; i<t; i++){
        long long int base = 1;
        scanf("%d %d",&n,&k);
        
        if (n == k) printf("%d\n",base);

        else{
            for (int j=n; j>k; j--){
                base *= j;

                if (base > 1000000007) base = base % 1000000007;
            }

            printf("%d\n",base);
        }
    }

    return 0;
}